wormhole-connect/src/views/v2/Bridge/WalletConnector/Sidebar.tsx

```diff
@@ -0,0 +1,192 @@
+  );
+
+  const renderWalletOptions = useCallback(
+    (wallets: WalletData[]): JSX.Element[] => {
+      const walletsSorted = wallets.sort((w) => (w.isReady ? -1 : 1));
+
+      const walletsFiltered = !search
+        ? walletsSorted
+        : walletsSorted.filter(
+            ({ name, type }: WalletData) =>
+              name.toLowerCase().includes(search) ||
+              type.toLowerCase().includes(search),
+          );
+
+      return walletsFiltered.map((wallet) => {
+        const isWalletReady = wallet.isReady;
+
+        return (
+          <ListItemButton
+            key={wallet.name}
+            dense
+            sx={{
+              display: 'flex',
+              flexDirection: 'row',
+            }}
+            onClick={() => {
+              if (isWalletReady) {
+                connect(wallet);
+              } else {
+                window.open(wallet.wallet.getUrl());
+              }
+            }}
+          >
+            <ListItemIcon>
+              <WalletIcon name={wallet.name} icon={wallet.icon} height={32} />
+            </ListItemIcon>
+            <Typography fontSize={14}>
+              <div className={`${!isWalletReady && classes.notInstalled}`}>
+                {!isWalletReady && 'Install'} {wallet.name}
+              </div>
+              <div className={classes.context}>{wallet.type.toUpperCase()}</div>
+            </Typography>
+          </ListItemButton>
+        );
+      });
+    },
+    [connect, search],
+  );
+
+  const sidebarContent = useMemo((): JSX.Element => {
+    switch (walletOptionsResult.state) {
+      case 'loading':
+        return <CircularProgress />;
+      case 'error':
+        return (
+          <AlertBanner show={true} content={walletOptionsResult.error} error />
+        );
+      case 'result':
+        if (walletOptionsResult.options?.length === 0) {
+          return <></>;
+        }
+        return (
+          <>
+            <List>
+              <ListItem>
+                <Typography component={'div'} fontSize={16}>
+                  Connect a wallet
+                </Typography>
+              </ListItem>
+              <ListItem>
+                <TextField
+                  fullWidth
+                  placeholder="Search for a wallet"
+                  size="small"
+                  variant="outlined"
+                  onChange={(e) => {
+                    setSearch(e.target.value?.toLowerCase());
+                  }}
+                  InputProps={{
+                    startAdornment: (
+                      <InputAdornment position="start">
+                        <SearchIcon />
+                      </InputAdornment>
+                    ),
+                  }}
+                />
+              </ListItem>
+              {renderWalletOptions(walletOptionsResult.options)}
+            </List>
+          </>
+        );
+      default:
+        // TODO: Do we ever get to this case? If so, what should be the UI?
+        return <></>;
```

Comment:
i think this is fine because typescript would catch it if this were ever possible